openmc/deplete/coupled_operator.py

```diff
@@ -0,0 +1,529 @@
+    reduce_chain : bool, optional
+        If True, use :meth:`openmc.deplete.Chain.reduce` to reduce the
+        depletion chain up to ``reduce_chain_level``.
+
+        .. versionadded:: 0.12
+    reduce_chain_level : int, optional
+        Depth of the search when reducing the depletion chain. Only used
+        if ``reduce_chain`` evaluates to true. The default value of
+        ``None`` implies no limit on the depth.
+
+        .. versionadded:: 0.12
+
+    Attributes
+    ----------
+    model : openmc.model.Model
+        OpenMC model object
+    geometry : openmc.Geometry
+        OpenMC geometry object
+    settings : openmc.Settings
+        OpenMC settings object
+    dilute_initial : float
+        Initial atom density [atoms/cm^3] to add for nuclides that
+        are zero in initial condition to ensure they exist in the decay
+        chain. Only done for nuclides with reaction rates.
+    output_dir : pathlib.Path
+        Path to output directory to save results.
+    round_number : bool
+        Whether or not to round output to OpenMC to 8 digits.
+        Useful in testing, as OpenMC is incredibly sensitive to exact values.
+    number : openmc.deplete.AtomNumber
+        Total number of atoms in simulation.
+    nuclides_with_data : set of str
+        A set listing all unique nuclides available from cross_sections.xml.
+    chain : openmc.deplete.Chain
+        The depletion chain information necessary to form matrices and tallies.
+    reaction_rates : openmc.deplete.ReactionRates
+        Reaction rates from the last operator step.
+    burnable_mats : list of str
+        All burnable material IDs
+    heavy_metal : float
+        Initial heavy metal inventory [g]
+    local_mats : list of str
+        All burnable material IDs being managed by a single process
+    prev_res : Results or None
+        Results from a previous depletion calculation. ``None`` if no
+        results are to be used.
+    cleanup_when_done : bool
+        Whether to finalize and clear the shared library memory when the
+        depletion operation is complete. Defaults to clearing the library.
+    """
+    _fission_helpers = {
+        "average": AveragedFissionYieldHelper,
+        "constant": ConstantFissionYieldHelper,
+        "cutoff": FissionYieldCutoffHelper,
+    }
+
+    def __init__(self, model, chain_file=None, prev_results=None,
+                 diff_burnable_mats=False, normalization_mode="fission-q",
+                 fission_q=None, dilute_initial=1.0e3,
+                 fission_yield_mode="constant", fission_yield_opts=None,
+                 reaction_rate_mode="direct", reaction_rate_opts=None,
+                 reduce_chain=False, reduce_chain_level=None):
+        # warn of name change
+        warn(
+            "The Operator(...) class has been renamed and will "
+            "be removed in a future version of OpenMC. Use "
+            "CoupledOperator(...) instead.",
+            FutureWarning
+        )
```

Comment:
This warning should only be displayed if the user actually calls `Operator`. An easy way to do this is:
```Python
def Operator(*args, **kwargs):
    warn(...)
    return CoupledOperator(*args, **kwargs)
```